src/Microsoft.AspNetCore.Mvc.Razor.Host/Internal/TagHelperChunkVisitor.cs

```diff
@@ -67,27 +50,23 @@ public void Accept(Chunk chunk)
             var parentChunk = chunk as ParentChunk;
             var tagHelperChunk = chunk as TagHelperChunk;
 
-            if (parentChunk != null && !(parentChunk is TagHelperChunk))
-            {
-                Accept(parentChunk.Children);
-                return;
+            if (tagHelperChunk != null){
+                var viewComponentDescriptors = GetViewComponentDescriptors(tagHelperChunk);
+                if (viewComponentDescriptors.Any())
```

Comment:
Shouldn't need to `.Any()` here since the only thing in decorate is a foreach loop.